src/components/common/Extensions/NotificationBanner/NotificationBanner.tsx

```diff
@@ -87,16 +76,16 @@ const NotificationBanner: FC<PropsWithChildren<NotificationBannerProps>> = ({
           'ml-6 md:ml-2': !isAlt,
         })}
       >
-        {actionType && actionText && actionMessage && (
+        {action && (
           <>
-            {actionType === 'copy-url' && (
-              <CopyUrl actionText={actionMessage} />
+            {action.type === 'copy' && (
+              <CopyUrl actionText={action.copyContent} />
             )}
-            {actionType === 'redirect' && redirectUrl && (
-              <Link to={redirectUrl}>{actionMessage}</Link>
+            {action.type === 'redirect' && (
+              <Link to={action.href}>{actionMessage}</Link>
             )}
-            {actionType === 'call-to-action' && (
-              <button type="button" onClick={onClick}>
+            {action.type === 'call-to-action' && (
+              <button type="button" onClick={action.onClick}>
                 {actionMessage}
               </button>
             )}
```

Comment:
I think you can just use switch here like

```
{(() => {
  switch (action?.type) {
    case "copy-url": {
       return <CopyUrl actionText={action.copyContent} />
    }
    ...(rest of the cases here)...
    default:
      return null;
  }
})()}
```

Reply:
Fixed